app/models/waste_carriers_engine/transient_registration.rb

```diff
@@ -114,6 +113,34 @@ def pending_manual_conviction_check?
       renewal_application_submitted? && conviction_check_required?
     end
 
+    def can_be_renewed?
```

Comment:
I think we maybe need a little more differentiation between `can_be_renewed?` and `ready_to_complete?`. Maybe some more specific naming would help?

My understanding is that `can_be_renewed?` should return true when it's OK for the user to proceed with the renewal process, as opposed to the transient_registration actually being renewable at that time. For that matter, it's not the transient_registration that renews, but the registration.

Whereas `ready_to_complete?` should only return true when the transient_registration is in such a state that renewal can now actually occur.

Is my understanding correct here?

Reply:
Yes. Where I got a bit stumped was that we referred to the original state machine `:renew` event on the metaData object for every transition because its called in the base `FormsController` (`metaData.may_renew?`). This includes when you do the GET request on the Renewal start page.

So ideally in the GET request for that page I'd want to call `can_be_renewed?` on the registration object. Then for every subsequent page i.e. when we have a saved transient_registration object, I'd want to call something like `can_progress_renewal?` on the transient_registration object.

Moving where we do the checks is one thing, but I felt like a refactor like that was a step to far for the scope of this PR. And in light of your main comment might have been wasted as well.

So we didn't really differentiate the checks before (we just called `may_renew?` in both cases) so I'm happy to leave as is. I think your main point about different state machines and transitions thought for the 2 registrations is where we should focus our efforts.